src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -67,6 +80,50 @@ public FilterPanel(UI ui, IModel model, PanelControl parentPanelControl, int pan
 
         filterTextField.setKeywords(all);
     };
+    
+    private Node createNameBar() {
+        nameText = new Text(panelName);
+    
+        renameButton = new Label(RENAME_PANEL);
+        renameButton.getStyleClass().add("label-button");
+        renameButton.setId(model.getDefaultRepo() + "_col" + panelIndex + "_renameButton");
+        renameButton.setOnMouseClicked(e -> {
```

Comment:
Try to separate the event handlers from the code which creates the elements. `createNameBar` should call another method which creates all the elements, and all it should do after that is bind the event handler. This will facilitate refactoring the views in future.